phoenix-core-client/src/main/java/org/apache/phoenix/schema/tuple/ResultTuple.java

```diff
@@ -101,4 +101,16 @@ public boolean getValue(byte[] family, byte[] qualifier,
         ptr.set(kv.getValueArray(), kv.getValueOffset(), kv.getValueLength());
         return true;
     }
+
+    @Override
+    public long getKeyValueBytesSize() {
+        if (result == null || result.isEmpty()) {
+            return 0;
+        }
+        long totalSize = 0;
+        for (Cell cell : result.rawCells()) {
```

Comment:
nit: There seems to be a utility method - getTotalSizeOfCells on the Result class

Reply:
I tried that but it has lot of overhead values added, i see quite a diff in the cell values. It is using some additional heap bytes fixed overhead which is not serialized size that client application needs to worry about for memory tracking purpose.